scalasql/core/src/DbApi.scala

```diff
@@ -522,7 +522,8 @@ object DbApi {
         if (savepointStack.lastOption.exists(_ eq savepoint)) {
           // Only release if this savepoint has not been rolled back,
           // directly or indirectly
-          connection.releaseSavepoint(savepoint)
+          // TODO - is this necessary? Not all jdbc drivers support releaseSavepoint (MsSql)
+          // connection.releaseSavepoint(savepoint)
```

Comment:
@lihaoyi what do you think about this?

Reply:
If  the underlying JDBC driver doesn't support it, there's nothing we can do, and we can just conditionally disable that line. But we should make sure we understand and document what the consequences are: does SqlServer not support savepoints at all? Or can you create them but not release them? So what does that mean for users?

Reply:
I don't know why no tests failed when commented out, I thought we did have pretty good tests for a variety of transaction/savepoint scenarios in https://github.com/com-lihaoyi/scalasql/blob/main/scalasql/test/src/api/TransactionTests.scala, but maybe I missed a spot

Reply:
> I can try to implement that conditional call to `releaseSavepoint()`, but I suspect at the time I faced this it was not trivial.

I revisited this today, and indeed I can't just write `dialect.isInstanceOf[MsSqlDialect]` because in `core` I don't have access to `MsSqlDialect`. 

Shall I add an `def supportSavepointRelease: Boolean` to `DialectConfig`? Shall I add a try/catch and swallow the exception?

Reply:
let's go with `def supportSavepointRelease: Boolean`

Reply:
done in 63c538a548ad56a0f484902cd113d598ae4b8b4c